lib/charms/hpc_libs/v0/slurm_ops.py

```diff
@@ -240,6 +192,175 @@ def get_options(self, *keys: str) -> Mapping[str, Any]:
 
         return configs
 
+    @abstractmethod
+    def get(self, key: Optional[str] = None) -> Any:
+        """Get specific configuration value for Slurm component."""
+
+    @abstractmethod
+    def set(self, config: Mapping[str, Any]) -> None:
+        """Set configuration for Slurm component."""
+
+    @abstractmethod
+    def unset(self, *keys: str) -> None:
+        """Unset configuration for Slurm component."""
+
+
+class MungeKeyManager(ABC):
+    """Control the munge key."""
+
+    def get(self) -> str:
+        """Get the current munge key.
+
+        Returns:
+            The current munge key as a base64-encoded string.
+        """
+
+    def set(self, key: str) -> None:
+        """Set a new munge key.
+
+        Args:
+            key: A new, base64-encoded munge key.
+        """
+
+    def generate(self) -> None:
+        """Generate a new, cryptographically secure munge key."""
```

Comment:
Since this is a abstract class, shouldn't these methods be decorated with the `abstractmethod` decorator? That way Python will complain if we don't implement the `get`, `set`, and `generate` methods on inheriting objects:

```suggestion
class MungeKeyManager(ABC):
    """Control the munge key."""

    @abstractmethod
    def get(self) -> str:
        """Get the current munge key.

        Returns:
            The current munge key as a base64-encoded string.
        """

    @abstractmethod
    def set(self, key: str) -> None:
        """Set a new munge key.

        Args:
            key: A new, base64-encoded munge key.
        """
   
   @abstractmethod
    def generate(self) -> None:
        """Generate a new, cryptographically secure munge key."""
```